c10/core/DispatchKey.h

```diff
@@ -291,6 +319,35 @@ C10_API std::ostream& operator<<(std::ostream&, DispatchKey);
 // torch::dispatch(torch::kCPU, ...) is also valid.
 constexpr DispatchKey kAutograd = DispatchKey::Autograd;
 
+// Check if a DispatchKey is an alias mapping to other runtime keys.
+static inline bool isAliasDispatchKey(DispatchKey k) {
+  return k > DispatchKey::NumDispatchKeys && k < DispatchKey::NumDispatchKeysWithAlias;
+}
+
+static inline std::array<DispatchKey, 7> resolveAliasAutogradKey() {
```

Comment:
Yeah, `std::array<DispatchKey, 7>` type here is weird (although, it is true that it is the same size as a DispatchKeySet!)

Reply:
I guess you didn't do `DispatchKeySet` here because you don't have `DispatchKeySet` defined at this point in time. It's probably not a big deal if you just put these definitions in `DispatchKeySet.h`

Reply:
Haha yea, the main reason I put it here is mainly that our DispatchKeySet is not a iterator-like object. Per Ed's comment above, it's worth adding it to DispatchKeySet now so I'll probably do that ;)